hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/CallFeature.java

```diff
@@ -291,12 +357,325 @@ public void ethCallReentrancyCallFunction() {
+                .tokenId();
+        var data = encodeData(PRECOMPILE, PAUSE_GET_STATUS_UNPAUSE_GET_STATUS, asAddress(tokenId));
+        var response = callContract(data, precompileContractAddress);
+        var statusAfterPause = response.getResult().substring(2, 66);
+        var statusAfterUnpause = response.getResult().substring(66);
+
+        // pause status is true
+        assertThat(Integer.valueOf(statusAfterPause, 16)).isEqualTo(1);
+        // unpause status is false
+        assertThat(Integer.valueOf(statusAfterUnpause, 16)).isEqualTo(0);
+    }
+
+    // ETHCALL-077, ETHCALL-078
+    @Then("I call function that freezes {string} token gets freeze status unfreezes and gets freeze status")
+    public void ethCallFreezeFungibleGetFreezeStatusUnfreezeGetFreezeStatus(String tokenName) {
+        var tokenId = tokenClient
+                .getToken(TokenClient.TokenNameEnum.valueOf(tokenName))
+                .tokenId();
+        var data = encodeData(
+                PRECOMPILE, FREEZE_TOKEN_GET_STATUS_UNFREEZE_GET_STATUS, asAddress(tokenId), asAddress(admin));
+        var response = callContract(data, precompileContractAddress);
+        var statusAfterFreeze = response.getResult().substring(2, 66);
+        var statusAfterUnfreeze = response.getResult().substring(66);
+
+        // status after freeze is true
+        assertThat(Integer.valueOf(statusAfterFreeze, 16)).isEqualTo(1);
+        // status after unfreeze is false
+        assertThat(Integer.valueOf(statusAfterUnfreeze, 16)).isEqualTo(0);
+    }
+
+    // ETHCALL-079
+    @Then("I call function that approves FUNGIBLE token and gets allowance")
+    public void ethCallApproveFungibleTokenGetAllowance() {
+        var data = encodeData(
+                PRECOMPILE,
+                APPROVE_TOKEN_GET_ALLOWANCE,
+                asAddress(fungibleTokenId),
+                asAddress(receiverAccountAlias),
+                new BigInteger("1"),
+                new BigInteger("0"));
+
+        var response = callContract(data, precompileContractAddress);
+        var allowance = response.getResult().substring(2, 66);
+
+        // allowance should equal amount
+        assertThat(new BigInteger(allowance)).isEqualTo(BigInteger.valueOf(1L));
+    }
+
+    // ETHCALL-080
+    @Then("I call function that approves NFT token and gets allowance")
+    public void ethCallApproveNFTTokenGetAllowance() {
+        var data = encodeData(
+                PRECOMPILE,
+                APPROVE_TOKEN_GET_ALLOWANCE,
+                asAddress(nonFungibleTokenId),
+                asAddress(receiverAccountAlias),
+                new BigInteger("0"),
+                new BigInteger("2"));
+        var response = callContract(data, precompileContractAddress);
+        var approvedAddress = response.getResult().substring(66);
+
+        // approved address should equal the spender
+        assertThat(asAddress(approvedAddress)).isEqualTo(asAddress(receiverAccountAlias));
```

Comment:
`asAddress` usually is used when passing data for headlong encoding. You can achieve the same result by left padding the receiverAccountAlias with 0's using util.to32BytesString. Comment applies for the same places below.
```suggestion
        assertThat(approvedAddress).isEqualTo(to32BytesString(receiverAccountAlias));
```